File: src/business-rules/customers/handlers/getbyid-customer.handler.ts
import {
  ICustomerRepository,
  ICustomerRepositoryName,
} from '@domain/repositories';
import { Inject, Injectable, NotFoundException } from '@nestjs/common';
import { IGetByIdCustomerRequest } from '../requests';
import { ICustomerResponse } from '../responses';
import { IGetByIdCustomerHandler } from './igetbyid-customer.handler';

@Injectable()
export class GetByIdCustomerHandler implements IGetByIdCustomerHandler {
  constructor(
    @Inject(ICustomerRepositoryName) private repository: ICustomerRepository,
  ) {}

  async execute(request: IGetByIdCustomerRequest): Promise<ICustomerResponse> {
    const customer = await this.repository.findById(request.id);
    if (!customer) {
      throw new NotFoundException('Customer not found');
    }

    const response: ICustomerResponse = {
      payload: {
        id: customer.id,
        name: customer.name,
        email: customer.email,
        birthDate: customer.birthDate,
      },
    };

    return response;
  }
}

File: dotnet/BusinessRules/Customers/Handlers/GetByIdCustomerHandler.cs
using dotnet.BusinessRules.Customers.Requests;
using dotnet.BusinessRules.Customers.Responses;
using dotnet.BusinessRules.Exceptions;
using dotnet.Domain.Repositories;

namespace dotnet.BusinessRules.Customers.Handlers
{
    public class GetByIdCustomerHandler : IGetByIdCustomerHandler
    {
        private readonly ICustomerRepository _customerRepository;

        public GetByIdCustomerHandler(
            ICustomerRepository customerRepository
        )
        {
            _customerRepository = customerRepository;
        }

        public CustomerResponse ExecuteAsync(GetByIdCustomerRequest request)
        {
            var customer = _customerRepository.FindById(request.Id);
            if (customer == null)
            {
                throw new NotFoundException("Cliente não encontrado");
            }

            return new CustomerResponse
            {
                Payload = new CustomerItemResponse
                {
                    Id = customer.Id,
                    Name = customer.Name,
                    Email = customer.Email,
                    BirthDate = customer.BirthDate
                }
            };
        }
    }
}